tools/mksurfdata_esmf/src/mkagfirepkmonthMod.F90

```diff
@@ -220,6 +225,7 @@ subroutine get_dominant_indices(dynamicMaskList, dynamicSrcMaskValue, dynamicDst
     rc = ESMF_SUCCESS
 
     if (associated(dynamicMaskList)) then
+       ! TODO: Can parts of this loop and/or its subloops be broken out into separate functions/subroutines to facilitate testing?
```

Comment:
This subroutine gets called in a way not familiar to me in this line:
`call ESMF_DynamicMaskSetR4R8R4(dynamicMask, dynamicMaskRoutine=get_dominant_indices,  &
handleAllElements=.true., rc=rc)`
So I could see the subroutine cascading to other subroutines or functions but not the subroutine itself broken up into multiple calls, if that makes sense.

Reply:
Resolving in preparation for merging PR.